st2common/st2common/persistence/base.py

```diff
@@ -112,12 +112,52 @@ def aggregate(cls, *args, **kwargs):
         return cls._get_impl().aggregate(*args, **kwargs)
 
     @classmethod
-    def add_or_update(cls, model_object, publish=True, dispatch_trigger=True):
+    def insert(cls, model_object, publish=True, dispatch_trigger=True,
+               log_not_unique_error_as_debug=False):
+        if model_object.id:
+            raise ValueError('id for object %s was unexpected.' % model_object)
+        try:
+            model_object = cls._get_impl().insert(model_object)
+        except NotUniqueError as e:
+            if log_not_unique_error_as_debug:
```

Comment:
Honestly, we could add the same argument to `add_or_update` method (without needing to add insert and update methods) and be done with it :P